src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Caching/WeakReferenceKey.cs

```diff
@@ -11,35 +11,28 @@ namespace NewRelic.Agent.Extensions.Caching
     /// <typeparam name="T"></typeparam>
     public class WeakReferenceKey<T> where T : class
     {
+        private readonly int _hashCode;
         private WeakReference<T> WeakReference { get; }
 
         public WeakReferenceKey(T cacheKey)
         {
             WeakReference = new WeakReference<T>(cacheKey);
+            _hashCode = cacheKey.GetHashCode(); // store the hashcode since we use it in the Equals method and the object could have been GC'd by the time we need to look for it
         }
 
         public override bool Equals(object obj)
         {
             if (obj is WeakReferenceKey<T> otherKey)
             {
-                if (WeakReference.TryGetTarget(out var thisTarget) &&
-                    otherKey.WeakReference.TryGetTarget(out var otherTarget))
-                {
-                    return ReferenceEquals(thisTarget, otherTarget);
-                }
+                return otherKey.GetHashCode() == _hashCode;
```

Comment:
Doesn't this risk a collision if you aren't checking `ReferenceEquals`?

Reply:
Well... I guess it's possible but in an LRU of 100 items of the same type, it seems unlikely that two different instances would have the same hashcode.

Reply:
See if you like it any better now.